AdyenComponents/Apple Pay/ApplePayComponentExtensions.swift

```diff
@@ -34,14 +34,114 @@ extension ApplePayComponent: PKPaymentAuthorizationViewControllerDelegate {
+                                       }
+            
+                                       if result.status == .success,
+                                          result.paymentSummaryItems.isEmpty == false {
+                                           do {
+                                               try self.applePayPayment.update(with: result.paymentSummaryItems)
+                                           } catch {
+                                               self.delegate?.didFail(with: error, from: self)
+                                           }
+                                       }
+                                       completion(result)
+                                   })
+    }
+
+    public func paymentAuthorizationViewController(_ controller: PKPaymentAuthorizationViewController,
+                                                   didSelect shippingMethod: PKShippingMethod,
+                                                   handler completion: @escaping (PKPaymentRequestShippingMethodUpdate) -> Void) {
+        guard let applePayDelegate = applePayDelegate else {
+            return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+        }
+
+        let applePayPayment = self.applePayPayment
+        applePayDelegate.didUpdate(shippingMethod: shippingMethod,
+                                   for: applePayPayment,
+                                   with: { [weak self] result in
+                                       guard let self = self else {
+                                           return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+                                       }
+
+                                       if result.status == .success,
+                                          result.paymentSummaryItems.isEmpty == false {
+                                           do {
+                                               try self.applePayPayment.update(with: result.paymentSummaryItems)
+                                           } catch {
+                                               self.delegate?.didFail(with: error, from: self)
+                                           }
+                                       }
+                                       completion(result)
+                                   })
+    }
+
+    @available(iOS 15.0, *)
+    public func paymentAuthorizationViewController(_ controller: PKPaymentAuthorizationViewController,
+                                                   didChangeCouponCode couponCode: String,
+                                                   handler completion: @escaping (PKPaymentRequestCouponCodeUpdate) -> Void) {
+        guard let applePayDelegate = applePayDelegate else {
+            return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+        }
+
+        let applePayPayment = self.applePayPayment
```

Comment:
`applePayPayment` is copied here then used in line 113/116 but not 122. is it intended?